src/locutus/model/terminology.py

```diff
@@ -176,6 +176,11 @@ def add_code(self, code, display, editor=None, description=None):
                 target="self",
                 new_value=code,
             )
+            self.add_provenance(

```

Comment:
I recommend you add in a comment above this prov explaining why you are adding a separate provenance message (clarify that this one goes into the code's provenance, whereas the other is going into the table/terminology's prov.